packages/lambda-powertools-pattern-basic/__tests__/supplement-csv.js

```diff
@@ -0,0 +1,71 @@
+const supplementCsv = require('../supplement-csv')
+
+describe('supplement csv', () => {
+  describe('when there are no existing items defined', () => {
+    it('should return blank string when nothing to add', () => {
+      const expected = ''
+
+      const actual = supplementCsv()
+
+      expect(actual).toEqual(expected)
+    })
+
+    it('should return csv string of key value pairs provided', () => {
+      const expected = 'key1:value1,key2:value2,key3:value3'
+      const inputs = {
+        additional: 'key1:value1,key2:value2,key3:value3'
+      }
+
+      const actual = supplementCsv(inputs)
+
+      expect(actual).toEqual(expected)
+    })
+  })
+
+  describe('when there are no items to add', () => {
+    it('should return existing csv string', () => {
+      const expected = 'key1:value1,key2:value2,key3:value3'
+      const inputs = {
+        existing: 'key1:value1,key2:value2,key3:value3'
+      }
+
+      const actual = supplementCsv(inputs)
+
+      expect(actual).toEqual(expected)
+    })
+  })
+
+  describe('when there is an existing csv string', () => {
+    it('should supplement with items to add', () => {
```

Comment:
this test is quite hard to follow, why split them up? why not just expect the output of supplementCSV to equal `key1:value1,key2:value2,key3:value3,key4:value4,key5:value5,key6:value6`?

Reply:
I'm not guaranteeing the order of the returned csv as it is fully reconstructed.

Although the order could be worked out, I don't think we should rely on the implementation details of the function under test, otherwise the test will become brittle.

Reply:
I see, how about using `expect(...).to.deepEqual(...)` and compare the actual and expected after we split them by `,`?

Reply:
I might be missing something here. As this is an array the order of the elements is important:

```
expect([1,2,3]).toEqual([2,1,3]) => fail
```